controller/src/main/java/org/jboss/as/controller/descriptions/DefaultNotificationDescriptionProvider.java

```diff
@@ -23,13 +26,15 @@
 public class DefaultNotificationDescriptionProvider implements DescriptionProvider {
 
     private final String notificationType;
+    private final Stability stability;
     private final ResourceDescriptionResolver descriptionResolver;
     private final DataValueDescriptor dataValueDescriptor;
 
-    public DefaultNotificationDescriptionProvider(final String notificationType,
+    public DefaultNotificationDescriptionProvider(final NotificationDefinition definition,
```

Comment:
This is public API so to be cautious we should add a constructor and deprecate the existing one.

Hmm, probably better is to not add or change a constructor here and just statically write DEFAULT in the description. Then deprecate the entire class and copy the code into a private class in NotificationDefinition, leaving this as temporary cruft. I don't see any hook that would allow a user to use any other class than this as the DescriptionProvider.

Reply:
Point taken.  I'll clean this up.